django/http/multipartparser.py

```diff
@@ -273,6 +274,14 @@ def parse(self):
                     # If this is neither a FIELD or a FILE, just exhaust the stream.
                     exhaust(stream)
         except StopUpload as e:
+            # Getting the location of the temporary file and removing it.
+            for handler in handlers:
+                if isinstance(handler, TemporaryFileUploadHandler):
+                    temp_location = handler.file.temporary_file_path()
+                    try:
+                        os.remove(temp_location)
+                    except FileNotFoundError as e:
+                        pass
```

Comment:
I don't think `multipartparser` should have any knowledge about `TemporaryFileUploadHandler`. 

Handlers should share a common interface that we can invoke on `StopUpload` instead.